### DISCUSSION TITLE
Web GUI Backup

### DISCUSSION THREAD

**Discussion by mskenderian-wps** [linuxserver/Heimdall#695] [Ideas]
Feature Request!

Implement Web GUI Backup Import/Export

**Comment by phil-hudson** [comment]
+1, it would be awesome if we could specify a backup location into the docker container, and essentially have some basic backup implemented

**Comment by justjake** [comment]
I want this feature too so I can automate setup from my configuration management system. 

Here’s my crack at a design write up for v1 of this feature: https://www.notion.so/jitl/Heimdall-Backup-Restore-design-7cb69703ffad4d6a8cb2e036e3bcd7c2

Let me know what you think. Once Kode or another maintainer reviews the design, I’ll get started on implementation. 
CC @KodeStar

**Comment by 2relativ** [comment]
Full Backup would be nice. But I personally would just like to see an Import/export feature in the GUI for the apps. Just like bookmarks in an browser. This way you can easily setup a new user.

**Comment by sherwinmartin** [comment]
just FYI, i had to move my heimdall install inside a docker container. i was able to find the database file. in docker, it's found in /config/www/app.sqlite. i stopped the container and got that file. then i created a new container in a different server and started it up to initialize everything. then i stopped the new container. copied the app.sqlite file to the new container and started it up. everything transferred over.

i'm not sure how the file structure is set up outside a docker container install though but hope my info helps you. it's not a web gui but it is possible

**Reply by HoboPapa** [reply]
Automated backups work with this method too. I learned that the /config/www/icons/ folder must also be saved if all custom images are to be restored. 
The method I use is to mount a volume in the container at /config/www. Then my docker host cron job can create an incremental tar ball and rsync modifications periodically without having to access the container. The tar ball contains all of the www folder so the container configuration can be restored as needed. Heimdall does not change too often!

I also agree with mskenderian-wps that a method to backup and restore via the GUI would be awesome so that it can be accomplished manually. A possible method could be to tar the /config/www folder within the application and download/upload it through the browser.
I've only restored the volume when the container is stopped so the application would likely need a restart function after the config data is restored.

The idea presented by justjake is more complete than this idea. In retrospect mine may be a bit too scrappy for a polished feature. But it works for me because there is only one user. My tarball size is 2.1MB.  Your mileage may vary.

**Reply by cristian-ivanovici** [reply]
Tested on version 2.5.6 and is not working anymore.
Tested on version 2.4.13 and is working.

Something was changed on /data/config/www/app.sqlite structure or data from 2.4.13 to 2.5.6?

In version 2.5.6 I see an Import/Export button on Settings page but how to export from 2.4.13 if there is no Export button?

The only way that is working now is this:
On old device
1 have a configuration in /data/config/www/ from version 2.4.13, copy all from that folder to a backup folder
On new device
2 install to another device version 2.4.13, 
3 stop container with 2.4.13
4 copy from backup folder to new device /data/config/www
5 start container on new device with version 2.4.13
6 remove container
7 install container with explicit version EX: image: lscr.io/linuxserver/heimdall:2.5.6
8 now browse to heimdall web app, everything should work good
